src/aws/flb_aws_util.c

```diff
@@ -309,6 +310,11 @@ struct flb_http_client *request_do(struct flb_aws_client *aws_client,
         goto error;
     }
 
+    /* Increase the maximum HTTP response buffer size to fit large responses from AWS services */
+    if (flb_http_buffer_size(c, FLB_MAX_AWS_RESP_BUFFER_SIZE)) {
```

Comment:
So there are two problems here:
1. this function returns -1 when it fails and that actually evaluates to true IIRC
2. also, fluent bit code style says you can't call a function inside an if condition

So you need to do:

```
ret = flb_http_buffer_size(c, FLB_MAX_AWS_RESP_BUFFER_SIZE);
if (ret == -1) {
```